R/rand_corr_matrix_func.R

```diff
@@ -0,0 +1,48 @@
+#' Generates a random correlation matrix.
+#'
+#' @param corr_function_params Structured as follows : {n_sites, seed}.
+#' Ensure that this list follows this structure. Number of sites n_sites, and
+#' seed desired for rng.  Defaults are set to NULL.
```

Comment:
Add a bit more of a description of what will be returned, which will be (correct me here if I am wrong): 
- a symmetric matrix with off diagonal upper triangular values sampled randomly from -1 to 1 (and therefore lower triangular values that are the same). I am probably not saying this correctly in math speak.
- any considerations for making this tunable to how far from 0 the randomly sampled values are? Think its fine to leave as is but just a consideration if we did want to tune the off diagonal correlation strength

Reply:
This was a quick function that I had put together, but yes you are correct with that description.  And the cool thing about how the `generate_simulated_data`'s spatial process is done is that we can make another correlation function that does other things we want it to do.  One thing I was going to add when looking for scenarios was a function that allows me to pass in my own correlation matrix.

Reply:
^ Think that's a really good idea!